test/pylib/rest_client.py

```diff
@@ -5,132 +5,115 @@
-        resp = await self.get(resource_uri)
-        return await resp.text()
-
-    async def post(self, resource_uri: str, params: Optional[dict[str, str]]) \
-            -> aiohttp.ClientResponse:
-        """Post to remote resource or raise"""
-        resp = await self.session.post(resource_uri, params=params)
-        if resp.status != 200:
-            text = await resp.text()
-            raise RuntimeError(f"status code: {resp.status}, body text: {text}, "
-                               f"resource {resource_uri} params {params}")
-        return resp
-
-    async def put_json(self, resource_uri: str, json: dict) \
-            -> aiohttp.ClientResponse:
-        """Put JSON"""
-        resp = await self.session.request(method="PUT", url=resource_uri, json=json)
-        if resp.status != 200:
-            text = await resp.text()
-            raise RuntimeError(f"status code: {resp.status}, body text: {text}, "
-                               f"resource {resource_uri} json {json}")
-        return resp
-
-    async def delete(self, resource_uri: str) -> aiohttp.ClientResponse:
-        resp = await self.session.delete(url=resource_uri)
-        if resp.status != 200:
-            text = await resp.text()
-            raise RuntimeError(f"status code: {resp.status}, body text: {text}")
-        return resp
-
-
-class UnixRESTClient:
+        assert method in ["GET", "POST", "PUT", "DELETE"], f"Invalid HTTP request method {method}"
+        assert response_type is None or response_type in ["text", "json"], \
+                f"Invalid response type requested {response_type} (expected 'text' or 'json')"
+        # Build the URI
+        port = port if port else self.default_port if hasattr(self, "default_port") else None
+        port_str = f":{port}" if port else ""
+        assert host is not None or hasattr(self, "default_host"), "_fetch: missing host for " \
+                "{method} {resource}"
+        host_str = host if host is not None else self.default_host
+        uri = self.uri_scheme + "://" + host_str + port_str + resource
+        logging.debug(f"RESTClient fetching {method} {uri}")
+
+        async with request(method, uri,
+                           connector = self.connector if hasattr(self, "connector") else None,
+                           params = params, json = json) as resp:
+            if resp.status != 200:
+                text = await resp.text()
+                raise RuntimeError(f"status code: {resp.status}, body text: {text}, "
+                                   f"resource {uri} params {params} json {json}")
+            if response_type is not None:
+                # Return response.text() or response.json()
+                return await getattr(resp, response_type)()
+        return None  # Unreached
+
+    async def get(self, resource_uri: str, host: Optional[str] = None, port: Optional[int] = None,
+                  params: Optional[Mapping[str, str]] = None) -> None:
```

Comment:
Return type should be `Any`

Reply:
Fixed in https://github.com/scylladb/scylladb/commit/8c39b3b7dc7a3d956837eb340c75e00a5b4589a9.

Also added missing `return`.